api/types/entities.go

```diff
@@ -76,4 +76,10 @@ type OrderUpdateCreator struct {
 	FirstName string `json:"firstname"`
 	LastName  string `json:"lastname"`
 	Username  string `json:"username"`
-}
\ No newline at end of file
+}
+
+type UpdateSock struct {
```

Comment:
1. Remove `UpdateSock` from `entities.go` (not used)
2. Modify the `UpdateSockRequest` in `payloads.go` to look like this:

```go
type UpdateSockRequest struct {
	Sock     SockDTO          `json:"sock" validate:"required"`
	Variants []SockVariantDTO `json:"variants" validate:"required,dive"`
}
```

This is mainly so this endpoint is consistent with the create sock endpoint + you gain the ability to use some existing helper functions to convert from `SockDTO` to `Sock`.

Reply:
Ok understood